engine/runtime/src/main/scala/org/enso/compiler/PackageRepository.scala

```diff
@@ -230,8 +256,66 @@ object PackageRepository {
         pkg            = pkg,
         isLibrary      = true
       )
+      pkg
     }.toEither.left.map { error => Error.PackageLoadingError(error.getMessage) }
 
+    /** @inheritdoc */
+    override def initialize(): Either[Error, Unit] = {
+      val unprocessedPackages =
+        loadedPackages.keySet
+          .diff(loadedComponents.keySet)
+          .flatMap(loadedPackages(_))
```

Comment:
It's supposed to be run once, but it won't take effect on the second invocation. I added a synchronized block around the `unprotectedPackages` to guard it against the races.

Reply:
Can we guarantee that the initialization process (which may take a long time since it can even download new libraries) and which calls `resolveComponentGroups` will not race with other (synchronized with each other but not with init) calls of `resolveComponentGroups` coming from `ensurePackageIsLoaded`?

I.e. is `ensurePackageIsLoaded` guaranteed to be called only after `initialize` has finished?
Seems so, but I'd verify it and add a comment if we rely on it. Otherwise it may be worth synchronizing the whole block. If it is ran once it shouldn't affect performance anyway, right?

Reply:
Yes, it won't affect performance. I added synchronization on the whole init function